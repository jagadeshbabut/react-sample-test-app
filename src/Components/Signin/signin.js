import React, {Component, Fragment} from 'react';
import {Link} from 'react-router-dom';

class Signin extends Component {
    constructor(props) {
        super(props);
        this.email = React.createRef();
        this.password = React.createRef();
    }

    signin(event) {
        event.preventDefault();
        const user_info = JSON.parse(sessionStorage.getItem('user_info'));
        if (user_info.email === this.email.current.value && user_info.password === this.password.current.value) {
            this
                .props
                .history
                .push('/dashboard');
        }
    }

    render() {
        return (
            <Fragment>
                <form className="form-signin" onSubmit={(e) => this.signin(e)}>
                    <img
                        className="mb-4"
                        src="//getbootstrap.com/docs/4.3/assets/brand/bootstrap-solid.svg"
                        alt=""
                        width="72"
                        height="72"/>
                    <h1 className="h3 mb-3 font-weight-normal">Please sign in</h1>
                    <label htmlFor="inputEmail" className="sr-only">Email address</label>
                    <input
                        type="email"
                        id="inputEmail"
                        className="form-control"
                        placeholder="Email address"
                        ref={this.email}
                        required
                        autoFocus/>
                    <label htmlFor="inputPassword" className="sr-only">Password</label>
                    <input
                        type="password"
                        id="inputPassword"
                        className="form-control"
                        placeholder="Password"
                        ref={this.password}
                        required/>
                    <div className="checkbox mb-3">
                        <label>
                            <input type="checkbox" value="remember-me"/>
                            Remember me
                        </label>
                    </div>
                    <button className="btn btn-lg btn-primary btn-block" type="submit">Sign in</button>
                </form>
                <hr/>
                <p>
                    <Link to="/">Signup if new user!</Link>
                </p>
            </Fragment>
        )
    }
}

export default Signin;