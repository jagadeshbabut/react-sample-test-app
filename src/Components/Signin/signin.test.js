import React from 'react';
import {shallow} from 'enzyme';
import Signin from './signin';

describe('Signin Component', () => {
    it('Renders without crashing', () => {
        shallow(<Signin/>)
    });

    it('Signin component has not changed', () => {
        const wrapper = shallow(<Signin/>);
        expect(wrapper).toMatchSnapshot()
    });

    it('Check Goto Todos link is working', () => {
        const wrapper = shallow(<Signin/>);
        // expect(wrapper).
    });

});