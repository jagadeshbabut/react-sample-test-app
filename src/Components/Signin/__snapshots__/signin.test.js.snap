// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin Component Signin component has not changed 1`] = `
<Fragment>
  <form
    className="form-signin"
    onSubmit={[Function]}
  >
    <img
      alt=""
      className="mb-4"
      height="72"
      src="//getbootstrap.com/docs/4.3/assets/brand/bootstrap-solid.svg"
      width="72"
    />
    <h1
      className="h3 mb-3 font-weight-normal"
    >
      Please sign in
    </h1>
    <label
      className="sr-only"
      htmlFor="inputEmail"
    >
      Email address
    </label>
    <input
      autoFocus={true}
      className="form-control"
      id="inputEmail"
      placeholder="Email address"
      required={true}
      type="email"
    />
    <label
      className="sr-only"
      htmlFor="inputPassword"
    >
      Password
    </label>
    <input
      className="form-control"
      id="inputPassword"
      placeholder="Password"
      required={true}
      type="password"
    />
    <div
      className="checkbox mb-3"
    >
      <label>
        <input
          type="checkbox"
          value="remember-me"
        />
        Remember me
      </label>
    </div>
    <button
      className="btn btn-lg btn-primary btn-block"
      type="submit"
    >
      Sign in
    </button>
  </form>
  <hr />
  <p>
    <Link
      replace={false}
      to="/"
    >
      Signup if new user!
    </Link>
  </p>
</Fragment>
`;
